multi-child-strawman.md

```diff
@@ -0,0 +1,289 @@
+Strawman proposal: 
+
+WIP - DO NOT MERGE
+
+This is a proposal for adding support for "inline" child elements that can occur in `source` and `target` values.
+
+I have already built a proof-of-concept implementation for this that adds support for one type of inline element (XLIFF 1.2 `<ph></ph>` elements). However, because this involves making a (backwards-compatible) change to the library's object model, I wanted to open up a conversation about the object model first before writing a complete implementation.
+
+The added file "multi-child-strawman.md" is the text proposal. I added it as a file so that we can add comments and discussion on a per-line basis. All the other code changes are the work for the proof-of-concept implementation.
+
+# Proposal: Adding support for XLIFF inline elements
+
+This proposal is divided into three sections:
+
+1. Changes to the `resources.namespace.key.{source,target}` properties
+2. General structure for representing XLIFF inline elements
+3. Specifics on which elements are supported and how they are mapped between JS and XLIFF (and vice-versa)
+
+
+## Proposal: `source`/`target` value type changes
+### Background
+
+Currently this library's object model for XLIFF documents defines `source` and `target` values as strings:
+
+```
+{
+  "resources": {
+    "namespace1": {
+      "key1": {
+        "source": "Hello",
+        "target": "Hallo"
+      },
+    }
+  },
+  "sourceLanguage": "en-US",
+  "targetLanguage": "de-CH"
+}
+```
+
+In addition to text, the XLIFF specifications define multiple child tags that can be included in `<source>`/`<target>` tags:
+
+- [XLIFF 1.2 spec](http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#Specs_Elem_Inline)
+- [XLIFF 2.0 spec](http://docs.oasis-open.org/xliff/xliff-core/v2.0/xliff-core-v2.0.html#inlineCodes)
```

Comment:
Is xliff 2.1 specification the same as 2.0?

Reply:
That's a great question. I didn't actually realize that the XLIFF 2.1 spec had been release.

After looking over the 2.1 spec and comparing it to the 2.0 spec, I believe that for the point of this proposal they are the same.

Dirty details if you want them:

As a quick way of trying to identify changes I downloaded the two specs (2.0 and 2.1) and compared them using a diff tool. Surprisingly that was actually pretty effective.

From what I found, the notable changes to 2.1 (i.e. new sections, things that are more than just minor clarifications or notes) are:
- [Section 5.9 ITS Module](http://docs.oasis-open.org/xliff/xliff-core/v2.1/xliff-core-v2.1.html#ITS-module)
- [Appendix A Media Type Registration Template for XLIFF Version 2.0 and higher Versions](http://docs.oasis-open.org/xliff/xliff-core/v2.1/xliff-core-v2.1.html#mediaType) (2.0 Appendix A, "XML Schemas and Catalog Listings (Informative)" was removed)
- [Appendix B Machine Readable Validation Artifacts (Informative)](http://docs.oasis-open.org/xliff/xliff-core/v2.1/xliff-core-v2.1.html#coreValidation)